src/control/lib/control/pool_test.go

```diff
@@ -654,22 +795,184 @@ func TestControl_ListPools(t *testing.T) {
 		},
 		"one pool": {
 			mic: &MockInvokerConfig{
-				UnaryResponse: MockMSResponse("host1", nil,
-					&mgmtpb.ListPoolsResp{
+				UnaryResponseSet: []*UnaryResponse{
+					MockMSResponse("host1", nil, &mgmtpb.ListPoolsResp{
 						Pools: []*mgmtpb.ListPoolsResp_Pool{
 							{
-								Uuid:    common.MockUUID(),
+								Uuid:    common.MockUUID(1),
 								SvcReps: []uint32{1, 3, 5, 8},
 							},
 						},
+					}),
+					MockMSResponse("host1", nil, queryResp(1)),
+				},
+			},
+			expResp: &ListPoolsResp{
+				Pools: []*Pool{
+					{
+						UUID:            common.MockUUID(1),
+						ServiceReplicas: []system.Rank{1, 3, 5, 8},
+						TargetsTotal:    42,
+						TargetsDisabled: 17,
+						Usage:           expUsage,
 					},
-				),
+				},
+			},
+		},
+		"one pool; no scm in query response": {
+			mic: &MockInvokerConfig{
+				UnaryResponseSet: []*UnaryResponse{
+					MockMSResponse("host1", nil, &mgmtpb.ListPoolsResp{
+						Pools: []*mgmtpb.ListPoolsResp_Pool{
+							{
+								Uuid:    common.MockUUID(1),
+								SvcReps: []uint32{1, 3, 5, 8},
+							},
+						},
+					}),
+					MockMSResponse("host1", nil, queryRespNoScm),
+				},
+			},
+			expErr: errors.New("missing scm"),
+		},
+		"one pool; no nvme in query response": {
+			mic: &MockInvokerConfig{
+				UnaryResponseSet: []*UnaryResponse{
+					MockMSResponse("host1", nil, &mgmtpb.ListPoolsResp{
+						Pools: []*mgmtpb.ListPoolsResp_Pool{
+							{
+								Uuid:    common.MockUUID(1),
+								SvcReps: []uint32{1, 3, 5, 8},
+							},
+						},
+					}),
+					MockMSResponse("host1", nil, queryRespNoNvme),
+				},
+			},
+			expErr: errors.New("missing nvme"),
```

Comment:
I'm thinking it might be better to return what we have, rather than erroring out. It could help users debug if something really looks wrong.

Reply:
this is really a "should never happen" scenario so I think an error is appropriate

Reply:
this test is referring to a nil Nvme reference as opposed to an empty Nvme tier

Reply:
have updated the test naming